server/app/routers/chat_router.py

```diff
@@ -0,0 +1,74 @@
+from fastapi import APIRouter, HTTPException
+from pydantic import BaseModel
+import google.generativeai as genai
+from typing import List, Optional
+
+# Initialize router
+router = APIRouter(
+    # prefix="/chat",
+    # tags=["chat"]
+)
+
+# Pydantic models for request/response
+class ChatMessage(BaseModel):
+    role: str
+    content: str
+
+class ChatRequest(BaseModel):
+    messages: List[ChatMessage]
+    temperature: Optional[float] = 0.7
+
+class ChatResponse(BaseModel):
+    response: str
+
+# Configure Gemini (you'll need to set this with your API key)
+GOOGLE_API_KEY = "AIzaSyDiFc8i8q20kHjppWfETtwe_5wMuPXFfHI"
```

Comment:
Do not expose your API key here please
You can add it to your env file
Git repos are scanned for API keys, specially in public repos

Instead you can add setup steps in the README for everyone to setup their own API keys

Reply:
It will automatically be resolved when yo push changes in new commit. You don't need to resolve it manually.